tests/sentry/workflow_engine/migrations/test_0105_deduplicate_workflows.py

```diff
@@ -0,0 +1,447 @@
+                type=Action.Type.DISCORD,
+                config={
+                    "target_identifier": "#discord-channel",
+                    "target_display": "Example Discord",
+                    "target_type": 0,
+                },
+                data={"tags": "foo"},
+            ),
+        ],
+    },
+    {
+        # triggers + action filters
+        "triggers": [
+            DataCondition(
+                type=Condition.ASSIGNED_TO,
+                comparison={
+                    "target_type": "Team",
+                    "target_identifier": "user@sentry.io",
+                },
+                condition_result=True,
+            ),
+            DataCondition(
+                type=Condition.ISSUE_RESOLUTION_CHANGE,
+                condition_result=True,
+                comparison=1,  # GroupStatus.RESOLVED
+            ),
+        ],
+        "action_filters": [
+            DataCondition(
+                type=Condition.EVENT_ATTRIBUTE,
+                comparison={
+                    "attribute": "user.email",
+                    "match": MatchType.EQUAL,
+                    "value": "user@sentry.io",
+                },
+                condition_result=True,
+            ),
+            DataCondition(
+                type=Condition.TAGGED_EVENT,
+                comparison={
+                    "key": "foo",
+                    "match": MatchType.EQUAL,
+                    "value": "bar",
+                },
+                condition_result=True,
+            ),
+        ],
+    },
+    {
+        # many of all
+        "triggers": [
+            DataCondition(
+                type=Condition.ASSIGNED_TO,
+                comparison={
+                    "target_type": "Team",
+                    "target_identifier": "user@sentry.io",
+                },
+                condition_result=True,
+            ),
+            DataCondition(
+                type=Condition.ISSUE_RESOLUTION_CHANGE,
+                condition_result=True,
+                comparison=1,  # GroupStatus.RESOLVED
+            ),
+        ],
+        "action_filters": [
+            DataCondition(
+                type=Condition.EVENT_ATTRIBUTE,
+                comparison={
+                    "attribute": "user.email",
+                    "match": MatchType.EQUAL,
+                    "value": "user@sentry.io",
+                },
+                condition_result=True,
+            ),
+            DataCondition(
+                type=Condition.TAGGED_EVENT,
+                comparison={
+                    "key": "foo",
+                    "match": MatchType.EQUAL,
+                    "value": "bar",
+                },
+                condition_result=True,
+            ),
+        ],
+        "actions": [
+            Action(
+                type=Action.Type.EMAIL,
+                config={
+                    "target_identifier": "user@sentry.io",
+                    "target_type": 2,
+                },
```

Comment:
same here re: fallthrough type and I guess nit that you could store the repeated actions and data conditions in variables to reuse instead of redefining them